src/nvrpc/Interfaces.h

```diff
@@ -50,9 +50,12 @@ class IContext {
   static IContext* Detag(void* tag) { return static_cast<IContext*>(tag); }
 
  protected:
-  IContext() = default;
+  IContext() : m_MasterContext(this) {}
+  IContext(IContext* master) : m_MasterContext(master) {}
   void* Tag() { return reinterpret_cast<void*>(this); }
 
+  IContext* m_MasterContext;
```

Comment:
@GuanLuo - can you help me understand the use case for `m_MasterContext`?  I'd like to keep upstream nvRPC up-to-date.

Reply:
The original design of nvRPC uses the address of the context as "tag" for completion queue, and the context maintains its own inner state to keep track of what kind of event to be handled next, the life cycle must follow some pattern. 

But then the context can't handle multiple events at the same time, in the case of bidirectional streaming, we can't distinguish if an event is related to Read or Write. So for streaming context, I added two "StateContext" members (derived from IContext) as a way to provides different tags (their address) for handling different events, one for read and one for write.

m_MasterContext is just pointing to the actual streaming context, such that once we know what type of event it is, we call the corresponding functions in the actual context.

Reply:
I think I get it.  

`m_MasterContext` is the streams context; however, you have two internal states for reading and writing that you want to independently manage events.  The tags for each of these `StateContext::Tag()` are different, so it allows you to have those two different tags in the completion queue at the same time, even though, they both point back to the same BiDirectionalStreamingContext object.  

LifeCycles are always a bit a mind-warp until you figured out just what's going one.  Good work.

Reply:
Yes, that is exactly what it is doing.

Reply:
👍 very nice 

I had previously only using the streaming interface for batching, i.e. collecting inputs, compute, then returning a stream of outputs.

This is a great generalization for the more complicated case of independent BiDirectional Reads/Writes.

When you finalize your changes to the internal version of nvRPC, please upstream it to https://github.com/NVIDIA/tensorrt-laboratory/tree/master/tensorrt-laboratory/nvrpc

I'll build a bidirectional flatbuffers example to work out the kinks with the different Message objects.